### ISSUE TITLE
Throws error if there is an `undefined` entry in children array

### ISSUE THREAD

**Issue by wbern** [TylorS/snabbdom-selector#15]
```
function traverseVNode(vNode, f) {
    function recurse(currentNode, isParent, parentVNode) {
        const length = currentNode.children && currentNode.children.length || 0;
        for (let i = 0; i < length; ++i) {
             // check if the entry is defined here
            const { children } = currentNode;
            if (children && typeof children[i] !== 'string') {
                const child = children[i];
                recurse(child, false, currentNode);
            }
        }
        f(currentNode, isParent, isParent ? void 0 : parentVNode);
    }
```

You can get `undefined` in children by various means, and I don't think an error should be thrown here as Snabbdom itself simply treats undefined as an empty string.

In my case, I put an `undefined` variable into the template string using JSX.

```
<h1>{ name }</h1>
```

**Comment by wbern** [comment]
Here is my work-around (sets children to `[]` after iterating bottom to top).

```
    tempSelect(selector, vnode) {
        // traverse down to the deepest child
        if (vnode.children) {
            const childSearch = vnode.children.find((childNode) => {
                if (childNode) {
                    return this.findLegacyComponentInNode(selector, childNode);
                }
                return null;
            });

            if (childSearch) {
                return childSearch;
            }
        }

        // and check it (without its children, due to bug with snabbdom-select)
        if (vnode) {
            return select(selector, {...vnode, children: []});
        }

        return null;
    }
```

**Comment by atomrc** [comment]
I have the same issue (I think). 

My use case is kind of special, I am actually parsing a raw html string to snabbdom and then I  select inside of it.

To do that, I use `html-to-hyperscript` that works correctly but seems to generate an `undefined` children when there is a comment inside the html. When I try to select, I then get the `TypeError: currentNode is undefined`.

Do you think the issue is with https://github.com/ivan-kleshnin/html-to-hyperscript that shouldn't generate that undefined element, or with `snabbdom-select` that should handle that case?

**Comment by TylorS** [comment]
Sorry guys, I've not been very active with this project. I no longer use snabbdom. If anyone is interested in taking this over, just let me know 😄 

As for the issue itself, I'm not sure where the error really is. When I wrote this, snabbdom did not handle "falsy" children values such as `null` or `undefined`.  I think adding a quick check [here](https://github.com/TylorS/snabbdom-selector/blob/master/src/findMatches.ts#L50) for falsy values should be enough if anyone wants to make the PR 😄

**Comment by TylorS** [comment]
Additionally, if anyone is interested. My virtual-dom library (heavily inspired by snabbdom) [mostly-dom](https://github.com/TylorS167/mostly-dom) has much much better support for this. 

There is [`hasCssSelector`](https://github.com/TylorS167/mostly-dom#-hascssselectorcssselector-string-vnode-vnode-boolean) which tests the if a VNode has a given CSS selector, sort of like `Element.matches`, [`querySelector`](https://github.com/TylorS167/mostly-dom#-queryselectorcssselector-string-vnode-vnode-vnode--null) and [`querySelectorAll`](https://github.com/TylorS167/mostly-dom#-queryselectorallcssselector-string-vnode-vnode-arrayvnode) which behave as closely to the DOM as possible.

**Comment by atomrc** [comment]
No problem, thanks for replying :)

A quick check is what I had in mind, I'll make a PR this evening if I find the time to do so. 

Is there a way I can use mostly-dom with cycle.js?

**Comment by TylorS** [comment]
A PR would be great, thanks @atomrc 

> Is there a way I can use mostly-dom with cycle.js?

It would be a relatively straightforward port of `@cycle/dom` from `snabbdom` to`mostly-dom`, but I've not done that myself. The only integration I've done is with Motorcycle.

**Comment by TylorS** [comment]
Released in v1.1.2

Thanks @atomrc!

**Comment by TylorS** [comment]
Hmm, it seems the wrong version was in the package.json :/ 
It should have been 1.2.1 instead

**Comment by TylorS** [comment]
Fixed in 1.1.2 and 1.2.1 ;)

**Comment by atomrc** [comment]
Awesome thanks @TylorS :)
